### PR TITLE
[color] Add v4

### PR DESCRIPTION
**Pull request by yinm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Qix-/color/releases
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

color v4 is released, but @types/color is v3.
So, I added v4.

### CODE REVIEW

File: types/color/v3/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    a(): number;
+    a(val: number): Color;
+    b(): number;
+    b(val: number): Color;
+    keyword(): string;
+    keyword<V extends string>(val: V): Color<V>;
+    hex(): string;
+    hex<V extends string>(val: V): Color<V>;
+    hexa(): string;
+    hexa<V extends string>(val: V): Color<V>;
```
Comment (yinm) [inline types/color/v3/index.d.ts:66]: `hexa()` is not supported in color v3 ([It's added in v4.2.0](https://github.com/Qix-/color/releases/tag/4.2.0)).
I think it should be removed but it is breaking changes in @types/color v3.
Thus, I keep it.

Comment (jameswlane) [reply types/color/v3/index.d.ts:66]: If introducing a breaking change, we should preserve backward compatibility by creating a new directory `types/color/v3/` with copies of the existing type files. Then, make the necessary changes in `types/color/`, and update the package.json to reflect the new version v4.2.3 for the updated type definitions.

**Review by jameswlane — CHANGES_REQUESTED** [review]
As mentioned in the inline comment, we should create a new directory `types/color/v3/` to preserve the existing type definitions before introducing any breaking changes. Afterward, update the `package.json` to reflect the new version v4.2.3.

File: types/color/v3/index.d.ts (hunk 1 above)
Comment (yinm) [reply types/color/v3/index.d.ts:66]: @jameswlane 
I didn't introduce breaking changes in this PR.

I created a new directory `types/color/v3/` with copies of the existing type files (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/19c2b6809cf41c13a8ae5c97a9f04bec49ff6a5e) and update `.npmignore` and `package.json` for v4 (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/a77fe1068aedd7242bce058715aae10e14451d0f and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70988/commits/a2120f5246681b393e064f94354f440a39978a66). 

Are there any problems?

Thanks.

Comment (jameswlane) [reply types/color/v3/index.d.ts:66]: After further research, I found that the breaking change for v3 was introduced and approved by the color library developer two years ago. Given this context, the changes look good to me, and I’m approving them.

**Review by jameswlane — APPROVED** [review]

**Comment by yinm** [comment]
Ready to merge

**Pull request merged**